### DISCUSSION TITLE
Is it possible to use existing k8s secrets?

### DISCUSSION THREAD

**Discussion by btrepp** [goauthentik/authentik#2591] [Q&A] [answered]
I'm managing my k8s cluster using ansible and argocd. One of the key parts is that I'm strictly separating out 'secrets configuration' from app configuration. This allows me to automate more of the app side, eg bumping versions etc, as I don't need them to have secrets.

I've noticed the helm chart asks for passwords, is there a way to reference existing kubernetes secrets instead?. That way I can transfer them using something like sealed secrets or get the secrets in cluster a different way, without having to have them be near the non secret config, thus making it okay to store the non secret config in git.

**Comment by BeryJu** [comment, marked as answer]
You can use existing secrets or config maps, see https://github.com/goauthentik/helm/blob/main/charts/authentik/values.yaml#L117-L125

**Reply by btrepp** [reply]
Thanks!. That was super helpful. It did take me awhile but I got postgres and authentik using external secrets.

Code for anyone who is trying to do similar here
https://git.sr.ht/~btrepp/infrastructure/tree/main/item/projects/authentik.yml

**Reply by CaptainStealthy** [reply]
@btrepp I found this thread in a Google search - your sourcehut link returns a 404 error. Do you have another location where you can share this?

**Reply by CaptainStealthy** [reply]
Managed to figure this out. For anyone else who finds this thread, just add secret refs to `values.yaml` as outlined in the docs - as long as the env vars are named correctly, you can then omit those values (`authentik.postgresql.password`, for instance)

I also added the following to ensure that the postgres workload would pick up the existing secret. The link below was a big help.

```yaml
postgresql:
  enabled: true
# snip [...]
  existingSecret: authentik-secret
  secretKeys:
    adminPasswordKey: "postgresql-password"
    userPasswordKey: "postgresql-password"
```

ref - https://github.com/bitnami/charts/issues/29758

**Comment by hekimazuri** [comment]
Given the response from @btrepp
> It did take me awhile but I got postgres and authentik using external secrets.
and my own frustration in getting this to work, an example of how this is done would be great.
I really appreciate btrepp following up with a solution, but the link has gone bad and there is now no working example.

Seeing as this is a product designed to administer account security for enterprises, having the default example use k8s secrets objects rather than hard coding keys to the kingdom into a file that will likely end up in version control, would help global security posture, and decrease the likelyhood of a major breach of networks using your tool.
